tests/algorithms/image/threshold/test_local.py

```diff
@@ -227,8 +229,45 @@ def test_dispersion_extended_threshold(self):
         result4 = debug.final_mask()
         assert result2 == result4
 
+    def test_dispersion_electron_threshold(self):
+        from dials.algorithms.image.threshold import (
+            DispersionElectronThreshold,
+            DispersionElectronThresholdDebug,
+        )
+        from dials.array_family import flex
+
+        nsig_b = 3
+        nsig_s = 3
+        algorithm = DispersionElectronThreshold(
+            self.image.all(), self.size, nsig_b, nsig_s, 0, self.min_count
+        )
+        result1 = flex.bool(flex.grid(self.image.all()))
+        result2 = flex.bool(flex.grid(self.image.all()))
+        algorithm(self.image, self.mask, result1)
+        algorithm(self.image, self.mask, self.gain, result2)
+
+        debug = DispersionElectronThresholdDebug(
+            self.image, self.mask, self.size, nsig_b, nsig_s, 0, self.min_count
+        )
+        result3 = debug.final_mask()
+
+        assert result1.all_eq(result3)
+
+        debug = DispersionElectronThresholdDebug(
+            self.image,
+            self.mask,
+            self.gain,
+            self.size,
+            nsig_b,
+            nsig_s,
+            0,
+            self.min_count,
+        )
+        result4 = debug.final_mask()
+        assert result2 == result4
```

Comment:
Slightly more descriptive names than result[1234] might give the casual reader more insight what this test is actually testing. (This could also be encoded in the test name)